src/timestamp.cpp

```diff
@@ -0,0 +1,53 @@
+#include "g3log/time.hpp"
+#include "g3log/timestamp.hpp"
+#include <sstream>
+
+
+namespace g3 {
+
+
+	Timestamp::Timestamp(std::string& defaultFormat) :
```

Comment:
First things first – let’s nail down the data structure.

It is great that you pointed at timeval struct, I was not aware about it. It is exactly what I need: just two fields – time_t and microsecond part. I did a bit of a reading about sys/time.h (the header defining this structure) and it is not part of C++ standard library, but rather part of the POSIX standards:
http://stackoverflow.com/questions/25615571/how-to-use-sys-time-h-in-window
https://www.codefull.org/2015/12/systime-h-replacement-for-windows/

Yes, it means Windows support is limited. I checked your link and as I read it this struct is defined in weird Winsock2.h header which makes me think that functions that operate with this struct are totally different compared to POSIX.

So I think if you decide to go with timeval and sys/time.h then you are sacrificing Windows support (but I am not a Widows expert so I may be wrong).

So should it be custom Timestamp structure with time_t and factional part or should I use POSIX sys/time.h?